draft/error-handling.md

```diff
@@ -0,0 +1,148 @@
+
+```Go
+const errNotMoreGOTSpoilers = errors.New("No more GOT spoilers!")
+const errTargaryenFireImmunity = fmt.Errorf("%s Targaryen is immune to fire!", name)
+```
+
+## Errores personalizados
+
+También en Go podemos controlar los errores de una forma más personalizada. Para ello podemos crearnos nuestro propio *struct* que implemente la interface *error*, y así devolver nuestro error personalizado en el momento que queramos.
+
+```Go
+type ErrGOT struct {
+    message string
+    code int
+    ...
+}
+
+func NewErrGOT(message string, code int) *ErrGOT {
+    return &ErrGOT{
+        message: message,
+        code: code,
+    }
+}
+
+func (e *ErrGOT) Error() string {
+    return fmt.Sprintf("%s Error with code %d", e.message, e.code)
+}
+
+func translateToHighValyrian(word string) (string, error) {
+    if word == "Cersei will..." {
+        return "", NewErrGOT("No more GOT spoilers!", 1)
+    } else if word == "Daenerys will ..." {
+        return "", NewErrGOT("Daenerys Targaryen is immune to fire!", 2)
+    }
+    ...
+}
+
+func main() {
+    t, err := translateToHighValyrian(word)
+    if err != nil {
+        fmt.Println(err)
+    }
+
+    fmt.Printf("translate: %s", t)
+}
+```
+
+Esto nos daría el error *Daenerys Targaryen is immune to fire! Error with code 2*.
+
+Esta forma de crear errores personalizados se suele utilizar cuando necesitas errores más complejos con códigos de errores, marcas de tiempo, etc.
+
+## El futuro de los errores con Go 2
+
+Si eres como yo, y te gusta conocer lo que va a venir en el futuro, ahora vamos a hablar de las funcionalidades que se proponen en la nueva versión de Go, de forma concreta en las novedades que trae en el control de errores.
+
+Uno de los cambios que más están llamando la atención a los Gophers son las nuevas funciones que nos presentan para poder interactuar con los errores. La comunidad de Go piensa que lo mejor es usar errores personalizados, es por ello que estas nuevas funcionalidades van enfocadas en dar más herramientas para usarlos.
+
+Una de las cosas que nos encontramos es la nueva *interface Wrapper*, esta deben implementarlas nuestros errores custom para poder usar las nuevas funciones *IS* y *AS* que a continuación os explicaremos. La propuesta de la nueva versión nos dice que todo error que envuelva a otro error debe implementar está *interface*.
+
+```Go
+type Wrapper interface {
+        // Unwrap devuelve el siguiente error, en la cadena de errores.
+        // Si el error no esiste, Unwrap devolverá nil.
+        Unwrap() error
+}
+```
+
+Vamos a hablar de la función *IS* que nos permite seguir la cadena de errores llamando a la función Unwrap y busca si alguno coincide con el valor que se le pasa. Será usada para controlar los errores sentinelas (sentinel errores), que es un tipo de error con valor único. Cualquier error puede implementar el método *Is* para sobrescribir el comportamiento por defecto.
```

Comment:
El IS además de comprobar sobre sentinel errors, también sirve para comprobar sobre error types.

Reply:
Listo! Arreglado en el commit "fix error behaviour, AS and IS"